extensions/amp-a4a/amp-a4a-format.md

```diff
@@ -0,0 +1,358 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+
+# AMP A4A AD CREATIVE FORMAT
+
+** >> WORK IN PROGRESS.  SUBJECT TO CHANGE. << **
+
+_This set of standards is still in development and is likely to be revised.
+Feedback from the community is welcome.  Please comment here or on the [Intent
+to Implement](https://github.com/ampproject/amphtml/issues/4264)_.
+
+A4A (AMP Ads for AMPHTML Pages) is a mechanism for rendering fast, performant
+ads in AMP pages.  To ensure that A4A ad documents ("A4A creatives") can be
+rendered quickly and smoothly in the browser and do not degrade user experience,
+A4A creatives must obey a set of validation rules.  Similar in spirit to the
+[AMP format rules](../../spec/amp-html-format.md), A4A creatives have access to
+a limited set of allowed tags, capabilities, and extensions.
+
+## A4A Format Rules
+ 
+1. The creative must obey all rules given by the [AMP format rules](../.
+./spec/amp-html-format.md), included here by reference.  _*In addition*_:
+
+1. The creative must use `<html a4⚡>` or `<html a4amp>` as its enclosing tags.
+
+   _Rationale_: Allows validators to identify a creative document as either a 
+ general AMP doc or a restricted A4A doc and to dispatch appropriately.
+
+1. Media: Videos must not enable autoplay.
+
+  _Rationale_: Autoplay forces video content to be downloaded immediately, which
+ slows the page load.
+
+1. Media: Audio must not enable autoplay.
+
+  _Rationale_: Same as for video.
+
+1. Analytics: `<amp-analytics>` viewability tracking may only target the entire 
+ad.  In particular, it may not target any selectors for elements within the ad creative.
+
+  _Rationale_: In some cases, A4A may choose to render an ad creative in an 
+  iframe.  In those cases, host page analytics can only target the entire iframe anyway, and won’t have access to any finer-grained selectors.
+
+### Boilerplate
+
+A4A creatives use the same boilerplate as [general AMP documents
+do](https://github.com/ampproject/amphtml/blob/master/spec/amp-boilerplate.md)
+_except_ that they omit the `<noscript>` section:
+
+```
+<style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style>
```

Comment:
@Gregable Sorry yes, should have said timeout.  The rationale for removing it in the ads case is that we never even want to render an ad if the AMP runtime is missing: the degraded mode of render-without-all-the-AMP-functionality will break too many things, including all of the accounting mechanisms that make ads worth showing in the first place.  If the AMP runtime doesn't load, a blank rectangle is the right outcome.